### ISSUE TITLE
Slack Channel Override does not work

### ISSUE THREAD

**Issue by jescarri** [fjogeleit/policy-reporter#458]
The configuration schema supports setting the slack channel per each notification channel, like this:

````
slack:
  channels:
    - channel: kyverno-reports1
      customFields:
        cluster: Avocado
      filter:
        namespaces:
          include:
          - foo
          - bar
        policies:
          include:
          - test-policy
        priorities:
          include: warning
      name: team1
      skipExistingOnStartup: false
      sources:
      - kyverno
      webhook: SlackWebhook
````

However slack messages end up in the default channel of the webhook, I have traced the issue to the slack newPayload function, it does not set the slack channel in the payload.

I see the client config etc properly populating the channel and config read from the file.

I have it fixed by adding code like this:

`````
+       if s.channel != "" {
+               p.Channel = s.channel
+       }
`````

I can submit a patch if this is an issue and it's not a problem with my config file.

Thanks for the help!.

**Comment by fjogeleit** [comment]
Thanks for reporting, if you would provide a fix as PR I would appreciate it.

**Comment by jescarri** [comment]
Sweet, let me get the PR ready.

Thanks for your response.

**Comment by jescarri** [comment]
Hi @fjogeleit I found another bug regarding slack channels when using the a secret reference for the webhook configuration, I will submit the PR in a bit.

Thanks!

**Comment by fjogeleit** [comment]
Your changes were released with helm chart v24.0.1

thanks for your support
